rotkehlchen/db/calendar.py

```diff
@@ -0,0 +1,219 @@
+            to_ts: Timestamp | None = None,
+            addresses: list[BlockchainAddress] | None = None,
+            name: str | None = None,
+            description: str | None = None,
+            counterparty: str | None = None,
+            identifiers: list[int] | None = None,
+    ) -> 'CalendarFilterQuery':
+        if order_by_rules is None:
+            order_by_rules = [('timestamp', True)]
+
+        filter_query = cls.create(
+            and_op=and_op,
+            limit=limit,
+            offset=offset,
+            order_by_rules=order_by_rules,
+        )
+        filter_query.timestamp_filter = DBTimestampFilter(
+            and_op=True,
+            from_ts=from_ts,
+            to_ts=to_ts,
+        )
+
+        filters: list['DBFilter'] = [filter_query.timestamp_filter]
+        if addresses is not None:
+            filters.append(DBMultiStringFilter(
+                and_op=True,
+                column='address',
+                values=addresses,
+            ))
+        if name is not None:
+            filters.append(DBSubStringFilter(
+                and_op=True,
+                field='name',
+                search_string=name,
+            ))
+        if description is not None:
+            filters.append(DBSubStringFilter(
+                and_op=True,
+                field='description',
+                search_string=description,
+            ))
+        if counterparty is not None:
+            filters.append(DBSubStringFilter(
+                and_op=True,
+                field='counterparty',
+                search_string=counterparty,
+            ))
+        if identifiers is not None:
+            filters.append(DBMultiIntegerFilter(
+                and_op=True,
+                column='identifier',
+                values=identifiers,
+            ))
+
+        filter_query.filters = filters
+        return filter_query
+
+
+class DBCalendar:
+
+    def __init__(self, database: 'DBHandler') -> None:
+        self.db = database
+
+    def create_calendar_entry(
+            self,
+            name: str,
+            timestamp: int,
+            description: str,
+            counterparty: str | None,
+            address: BlockchainAddress | None,
+    ) -> int:
+        """Insert the provided event in the database.
+        May raise:
+        - InputError if any db constraint is not satisfied
+        """
+        with self.db.user_write() as write_cursor:
+            try:
+                write_cursor.execute(
+                    'INSERT INTO calendar(name, timestamp, description, counterparty, address) '
+                    'VALUES (?, ?, ?, ?, ?) RETURNING identifier',
+                    (
+                        name,
+                        timestamp,
+                        description,
+                        counterparty,
+                        address,
+                    ),
+                )
+            except sqlcipher.IntegrityError as e:  # pylint: disable=no-member
```

Comment:
Why integrity error? There is no foreign keys. So the only error I can see is already existing. Why not do `INSERT OR IGNORE` and then check if cursor write count is 1?

Reply:
It can't happen either since the primary key is an auto increasing integer

Reply:
> It can't happen either since the primary key is an auto increasing integer

But then we would have duplicate calendar events. Perhaps make title unique?